###############################################################################
Date: Tuesday 24 January 22:10:00 UTC 2023
Subject: With newer cmake-3.22, the FindGTest.cmake module is
       slightly differenct than older cmake-3.16 module. We don't have
       GTEST_INCLUDE_DIR set in latest cmake GTest module where as
       GTEST_LIBRARIES and GTEST_INCLUDE_DIRS are set on both older and new modules.
Source: Comcast
License: NA
Upstream-Status: Pending
Signed-off-by: Arjun <arjun_daasuramdass@comcast.com>
###############################################################################
--- a/unittests/CMakeLists.txt.orig	2023-01-24 08:28:22.536357264 +0000
+++ b/unittests/CMakeLists.txt	2023-01-24 08:31:39.383571780 +0000
@@ -26,7 +26,7 @@
 
 string(REPLACE "-Werror" "" CMAKE_CXX_FLAGS_DEBUG ${CMAKE_CXX_FLAGS_DEBUG})
 string(REPLACE "-Werror" "" CMAKE_CXX_FLAGS_RELEASE ${CMAKE_CXX_FLAGS_RELEASE})
-if (NOT GTEST_INCLUDE_DIR)
+if (NOT GTEST_LIBRARIES)
     message("Warning GTest wasn't found. GTest library will be built.")
     include(ExternalProject)
     #include(GoogleTest)
@@ -53,7 +53,7 @@
 
     include_directories("${source_dir}/googletest/include" "${source_dir}/googlemock/include")
 else ()
-    include_directories("${GTEST_INCLUDE_DIR}")
+    include_directories("GTEST_INCLUDE_DIRS")
 endif()
 
 # someone needs to figure out how to run this stuff under valgrind
@@ -102,7 +102,7 @@
   main.cpp)
 add_dependencies(rbus_gtest.bin rbus)
 
-if (NOT GTEST_INCLUDE_DIR)
+if (NOT GTEST_LIBRARIES)
     target_link_libraries(rbus_gtest.bin rbus libgtest libgmock gcov)
 else ()
     target_link_libraries(rbus_gtest.bin rbus gtest gcov)
